components/shared/net/lib.rs

```diff
@@ -940,6 +956,37 @@ pub enum CookieSource {
     NonHTTP,
 }
 
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct CookieChange {
+    changed: Vec<Serde<Cookie<'static>>>,
+    deleted: Vec<Serde<Cookie<'static>>>,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub enum CookieData {
+    Change(CookieChange),
+    Get(Vec<Serde<Cookie<'static>>>, String),
+    GetAll(Vec<Serde<Cookie<'static>>>),
+    Set(Result<(), ()>),
+    Delete(Result<(), ()>),
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct CookieAsyncResponse {
+    pub id: CookieRequestId,
+    pub event: CookieData,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub enum CookieRequestType {
```

Comment:
This type is currently not used. I was split between using a type like this one and wrapping the cookie store id and request id into one type and then only having one message type vs creating all the message types like I currently have.